release_dashboard/lib/logic/git.dart

```diff
@@ -0,0 +1,49 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:conductor_core/conductor_core.dart';
+
+// Allows empty string because the dart revision which is the only input using this regex could be empty
+final RegExp gitHashRegex = RegExp(r'^$|^[0-9a-f]{40}$');
+final String gitHashErrorMsg = 'Must be 40 alphanumeric characters or an empty string.';
+
+/// Supports multiple git hashes delimited by a comma or an empty string.
```

Comment:
This shouldn't be done in regex. Instead, rely on removing whitespace and splitting on the commas. Then you can loop over the parts against gitHashRegex